File: Bit_Manipulation/go/binary_to_string.go
package main


import (
  "fmt"
)


func getBinaryRepresentation(num float64) string {
  var binary string = "."
  var frac float64 = 0.5
  if num >= 1 || num <= 0 {
    return "ERROR"
  }

  for num > 0 {
    if len(binary) > 32 {
      return "ERROR"
    }

    if num >= frac {
      binary += "1"
      num -= frac
    } else {
      binary += "0"
    }

    frac /= 2
  }

  return binary
}


func main() {
  var num float64 = 0.72

  fmt.Println(getBinaryRepresentation(num))
}

File: Bit_Manipulation/python/binary_to_string.py
def get_binary_representation(num):
    binary = "."
    frac = 0.5

    if 0 <= num <= 1:
        return "ERROR"

    while num > 0:
        if len(binary) > 32:
            return "ERROR"

        if num >= frac:
            binary += "1"
        else:
            binary += "0"

        frac /= 2

    return binary


def main():
    num = 0.72

    print(get_binary_representation(num))


if __name__ == '__main__':
    main()
